storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -256,3 +260,33 @@ func (s *commitLogSource) Read(
 
 	return bootstrapResult, nil
 }
+
+func newReadCommitLogPredicate(
+	ns namespace.Metadata,
+	shardsTimeRanges result.ShardTimeRanges,
+	opts Options,
+) commitlog.ReadEntryPredicate {
+	return func(entryTime time.Time, entryDuration time.Duration) bool {
+		// Minimum and maximum times for which we want to bootstrap
+		shardMin, shardMax := shardsTimeRanges.MinMax()
+		shardRange := xtime.Range{
+			Start: shardMin,
+			End:   shardMax,
+		}
+
+		// How far into the past or future a commitlog might contain a write for a
+		// previous or future block
+		bufferPast := ns.Options().RetentionOptions().BufferPast()
+		bufferFuture := ns.Options().RetentionOptions().BufferFuture()
```

Comment:
Hey want to move all these vars to above so that they get calculated/called once?

e.g.
```go
		// Minimum and maximum times for which we want to bootstrap
		shardMin, shardMax := shardsTimeRanges.MinMax()
		shardRange := xtime.Range{
			Start: shardMin,
			End:   shardMax,
		}
 
		// How far into the past or future a commitlog might contain a write for a
		// previous or future block
		bufferPast := ns.Options().RetentionOptions().BufferPast()
		bufferFuture := ns.Options().RetentionOptions().BufferFuture()
		return func(entryTime time.Time, entryDuration time.Duration) bool {
			return xtime.Range{
				Start: entryTime.Add(-bufferPast),
				End:   entryTime.Add(entryDuration).Add(bufferFuture),
			}.Overlaps(shardRange)
		}
	}
```